### PR TITLE
Add Test Environment that runs tests w/out aiohttp

### PR DESCRIPTION
**Pull request by scbedd**
@seankane-msft  

This is related to [this issue](https://github.com/Azure/azure-sdk-for-python/issues/17516).

Can you poke holes in my logic? We don't really have anything to key off to identify one of these "optional" requirements that may or may not be present. So it's not really a case of extending `create_package_and_install.py` to identify additional requirements. 

To solve this generically without messing with anyone's day-to-day development, I created a new tox environment `whl_no_aio`. This is essentially a retread of `whl` (aka the default) tox environment. The only difference is that have an error-ignored line (notice the leading `-`) that attempts uninstall of aiohttp prior to invoking any of the install or test commands.

### CODE REVIEW

File: eng/pipelines/templates/jobs/archetype-sdk-client.yml [hunk 1]
```diff
@@ -178,7 +178,7 @@ jobs:
     - pwsh: |
         $toxenvvar = "whl,sdist"
         if ('$(System.TeamProject)' -eq 'internal') {
-          $toxenvvar = "whl,sdist,depends,latestdependency,mindependency"
+          $toxenvvar = "whl,sdist,depends,latestdependency,mindependency,whlnoaiohttp"
```
Comment (scbedd) [inline eng/pipelines/templates/jobs/archetype-sdk-client.yml:181]:
```suggestion
          $toxenvvar = "whl,sdist,depends,latestdependency,mindependency,whl_no_aio"
```

File: eng/tox/tox.ini [hunk 2]
```diff
@@ -79,6 +79,23 @@ commands =
...
+changedir = {toxinidir}
+deps =
+  {[base]deps}
+commands =
+    - {envbindir}/python -m pip uninstall aiohttp --yes
+    {envbindir}/python {toxinidir}/../../../eng/tox/create_package_and_install.py -d {envtmpdir} -p {toxinidir} -w {envtmpdir}
+    {envbindir}/python -m pip freeze
+    pytest \
+      {[testenv]default_pytest_params} \
+      --ignore-glob='*_async.py' \
```
Comment (scbedd) [inline eng/tox/tox.ini:94]: Not having this means that async tests implode with 

image

**Comment by seankane-msft** [comment]
Within the `create_package_and_install.py` file, can we add a flag for this toxenv where we add a clause like the following to verify that the uninstallation was successful:
```python
try:
    import aiohttp
    raise Exception("aiohttp was not successfully installed") 
except ImportError:
    pass
```

**Comment by scbedd** [comment]
I see what you're going after here. I'll add something like it.

**Comment by scbedd** [comment]
@seankane-msft  added!

File: eng/tox/try_import.py [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+# This script is used to execute bandit within a tox environment. Depending on which package is being executed against,
+# a failure may be suppressed.
+
+
+from subprocess import check_call, CalledProcessError
+import argparse
+import os
+import logging
+import sys
+import pdb
```
Comment (seankane-msft) [inline eng/tox/try_import.py:17]:
```suggestion
import argparse
import logging
```

Comment (seankane-msft) [reply eng/tox/try_import.py:17]: Nit: you don't need all these imports. Otherwise this is perfect.

**Review by seankane-msft — APPROVED** [review]
:shipit:

File: eng/tox/try_import.py (hunk 3 above)
Comment (scbedd) [reply eng/tox/try_import.py:17]: Big oof. Totally agree. Cleaned up comment, applied `black`, didn't clean up imports. /facepalm

**Pull request merged**